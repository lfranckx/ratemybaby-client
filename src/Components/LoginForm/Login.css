#login-form {
    position: relative;
    top: 45%;
    margin: 10px;
}

.input-box {
    margin: 10px;
}


#login-button {
    margin: 10px;
}