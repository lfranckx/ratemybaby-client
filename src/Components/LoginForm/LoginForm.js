import React, { Component } from 'react';
import AuthApiService from '../../Services/auth-api-service'
import BabyContext from '../../Contexts/BabyContext'

export default class LoginForm extends Component {
    static defaultProps = {
        onLoginSuccess: () => {}
    }

    static contextType = BabyContext

    state = { error: null }

    handleSubmitJwtAuth = ev => {
        ev.preventDefault()
        this.setState({ error: null })
        const { username, password } = ev.target

        AuthApiService.postLogin({
          username: username.value,
          user_password: password.value,
        })
        .then(res => {
            username.value = ''
            password.value = ''
            console.log('getUser response:', res);
            this.props.onLoginSuccess()
        })
        .catch(res => {
            this.setState({ error: res.error })
        })
    }

    render() {
        const { error } = this.state
        return (
                <form 
                    id='login-form'
                    onSubmit={this.handleSubmitJwtAuth}
                >
                    <div role='alert'>{error && <p className='error'>{error}</p>}</div>
                    <div className="input-box">
                        <label className='label'>
                            Username
                            {/* <span className='required'>*</span> */}
                        </label>
                        <input
                            type="text" 
                            name='username' 
                            className='username' />
                    </div>
                    <div className="input-box">
                        <label className='label'>
                            Password
                            {/* <span className='required'>*</span> */}
                        </label> 
                        <input 
                            type="password" 
                            name='password' 
                            className='password' />
                    </div>
                    <button id="login-button" type="submit">Login</button>
                </form>
        )
    }
}