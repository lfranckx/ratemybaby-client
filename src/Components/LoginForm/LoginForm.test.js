import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginForm from './LoginForm';

const simulateChangeOnInput = (wrapper, inputSelector, newValue) => {
    const input = wrapper.find(inputSelector);
    input.simulate('change', {
        target: {value: newValue}
    });
    return wrapper.find(inputSelector);
};

describe(`LoginForm component`, () => {
    const props = {
        handleLoginSuccess: () => {}
    };

    it(`renders form given props`, () => {
        const wrapper = shallow(<LoginForm {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    it(`lets me fill out the form and reset`, () => {
        const wrapper = shallow(<LoginForm {...props}/>);
        simulateChangeOnInput(wrapper, '#loginuser', 'test')
        simulateChangeOnInput(wrapper, '#loginpass', 3)
    });

    it(`lets me fill out the form and submit it`, () => {
        const wrapper = shallow(<LoginForm {...props}/>);
        const userInput = simulateChangeOnInput(wrapper, '#loginuser', 'test');
        const pwInput = simulateChangeOnInput(wrapper, '#loginpass', 'Test1234!');

        expect(userInput.props().value).toEqual('test');
        expect(pwInput.props().value).toEqual('Test1234!');

        const submitButton = wrapper.find('#login-button');
        submitButton.simulate('click')
    });
})