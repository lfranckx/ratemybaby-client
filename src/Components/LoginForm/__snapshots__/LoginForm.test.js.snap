// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders form given props 1`] = `
<form
  id="login-form"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="input-box"
  >
    <label
      className="label"
    >
      Username
    </label>
    <input
      aria-label="username"
      className="username"
      id="loginuser"
      name="username"
      required={true}
      type="text"
    />
  </div>
  <div
    className="input-box"
  >
    <label
      className="label"
    >
      Password
    </label>
    <input
      aria-label="password"
      autoComplete="on"
      className="password"
      id="loginpass"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    id="login-button"
    type="submit"
  >
    Login
  </button>
</form>
`;
